backend/tests/endpoints/test_message.py

```diff
@@ -92,74 +150,211 @@ async def test_send_message_successful(mock_get_user_room, mock_write, mock_cent
+                        "type": "unstyled",
+                    }
+                ],
+                "entityMap": {},
+            },
+            "room_id": "343235",
+            "saved_by": [],
+            "sender_id": "619ba4",
+            "threads": [],
+            "timestamp": 0,
+        },
+    }
+
+
+@pytest.mark.asyncio
+@mock.patch.object(DataStorage, "__init__", lambda x, y: None)
+async def test_update_message_empty_message(mock_dataStorage_read):
+    """Update message unsuccessful with an invalid message_id.
+
+    Args:
+        mock_dataStorage_read (AsyncMock): Asynchronous external api call
+    """
+    db = DataStorage("619ba4")
+    db.plugin_id = "34453"
+    mock_dataStorage_read.return_value = {}
+    response = client.put(update_message_test_url, json=update_message_test_payload)
+    assert response.status_code == 404
+    assert response.json() == {"detail": "Message not found"}
+
+
+@pytest.mark.asyncio
+@mock.patch.object(DataStorage, "__init__", lambda x, y: None)
+async def test_update_message_wrong_sender_id(mock_dataStorage_read):
+    """Update message unsuccessful with a wrong sender_id provided.
+
+    Args:
+        mock_dataStorage_read (AsyncMock): Asynchronous external api call
+    """
+    db = DataStorage("619ba4")
+    db.plugin_id = "34453"
+    fake_zc_core_message_data["sender_id"] = "6er34"
+    mock_dataStorage_read.return_value = fake_zc_core_message_data
+    response = client.put(update_message_test_url, json=update_message_test_payload)
+    assert response.status_code == 401
+    assert response.json() == {"detail": "You are not authorized to edit this message"}
+
+
+@pytest.mark.asyncio
+@mock.patch.object(DataStorage, "__init__", lambda x, y: None)
+async def test_update_message_check_status_code(
+    mock_dataStorage_read, mock_dataStorage_update
+):
+    """Update message unsuccessful when updating to zc core fails.
+
+    Args:
+        mock_dataStorage_read (AsyncMock): Asynchronous external api call
+        mock_dataStorage_update (AsyncMock): Asynchronous external api call
+    """
+    db = DataStorage("619ba4")
+    db.plugin_id = "34453"
+    fake_zc_core_message_data["sender_id"] = "619ba4"
+    mock_dataStorage_read.return_value = fake_zc_core_message_data
+    mock_dataStorage_update.return_value = {
+        "status_code": 422,
+        "message": "unprocessible error",
+    }
+    response = client.put(update_message_test_url, json=update_message_test_payload)
+    assert response.status_code == 424
+    assert response.json() == {
+        "detail": {"message not edited": mock_dataStorage_update.return_value}
+    }
+
+
+@pytest.mark.asyncio
+async def test_connection_timeout():
+    """Test for connection timeout"""
+    with mock.patch.object(
+        DataStorage, "__init__", Mock(return_value=exceptions.RequestException)
+    ) as mocker:
+        DataStorage.__init__("12334")
+        mocker.assert_called_with("12334")
```

Comment:
Can we move this to test_db.py?

Reply:
This can be moved there boss, then we will have to write test for the whole datastorage class, I am thinking we have indirectly written the test for the methods of the DataStorage inside the tests for each endpoints. If need be for us to write the test, no issue then, same way as above, instead of mocking only the init method, we mock the object and have access to all of the methods. @zxenonx

Reply:
Absolutely agree with you. 
We can mock the whole DataStorage then.

Reply:
Okay boss

Reply:
Okay boss